### PR TITLE
Add .gitattributes file to project [HZ-2901]

### PR DESCRIPTION
**Pull request by orcunc**
So that on Windows the .sh script files are preserved with LF line endings [HZ-2901]

[HZ-2901]: https://hazelcast.atlassian.net/browse/HZ-2901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# linux line-endings
+*.sh        text eol=lf
```
[the remark this answers is not in the document]
Comment (nishaatr) [reply .gitattributes:2]: personally I would prefer *.sh
I am on Windows so not sure what will happen if I open the files?
I have asked @orcunc if I can test this unless its been tested in Windows
Normally, when line endings are in Linux, Windows prompts me to convert. I have seen this before many times but not sure may be Windows can cope better nowadays

[the remark this answers is not in the document]
Comment (orcunc) [reply .gitattributes:2]: @nishaatr when git checks out if converts line endings for text files according to your OS. It was converting .sh files to have CLRF (\r\) ending and it was causing the error. With .gitattributes we are telling git not to convert .sh files but preserve the LF (\n) ending.  After this PR is merged delete your local repo and check out again. From IDE or editor check that the .sh files have LF ending. 

To test this PR I don't know any git command to change line ending for already checked out file.  But you can change the .sh files line endings manually and run docker build ... command again

[the remark this answers is not in the document]
Comment (nishaatr) [reply .gitattributes:2]: Sounds good @orcunc . I will revert my change, pull and test and let you know here

[the remark this answers is not in the document]
Comment (nishaatr) [reply .gitattributes:2]: actually @orcunc I presume you tested this in Windows?

[the remark this answers is not in the document]
Comment (orcunc) [reply .gitattributes:2]: Yes I tested on windows. For testing 
1. you need to change the .sh files (there are two files one for OSS one for EE ) to LF 
2. Run docker build ...
Both builds succeed

Then *after* the PR is merged. You need to delete the local repo and check out once more. You should see the .sh files with LF ending.

[the remark this answers is not in the document]
Comment (nishaatr) [reply .gitattributes:2]: great. thanks @orcunc 
i will test after you push the changes  
Approving shortly

**Review by nishaatr — APPROVED** [review]

**Review by ldziedziul — APPROVED** [review]

**Pull request closed without merging**